lib/mailer.service.ts

```diff
@@ -86,23 +88,28 @@ export class MailerService {
     /** Transporters setup **/
     if (mailerOptions.transports) {
       Object.keys(mailerOptions.transports).forEach((name) => {
-        this.transporters.set(
-          name,
-          this.transportFactory.createTransport(
-            this.mailerOptions.transports![name],
-          ),
-        );
-        this.initTemplateAdapter(this.templateAdapter, this.transporters.get(name)!);
+        const transporter = this.transportFactory.createTransport(this.mailerOptions.transports![name])
+        this.transporters.set(name, transporter);
+        this.verifyTransporter(transporter, name);
+        this.initTemplateAdapter(this.templateAdapter, transporter);
       });
     }
 
     /** Transporter setup **/
     if (mailerOptions.transport) {
       this.transporter = this.transportFactory.createTransport();
+      this.verifyTransporter(this.transporter);
       this.initTemplateAdapter(this.templateAdapter, this.transporter);
     }
   }
 
+  private verifyTransporter(transporter: Transporter, name?: string): void {
+    const transporterName = name ? ` '${name}'` : '';
+    transporter.verify()
+      .then(() => this.mailerLogger.error(`Transporter${transporterName} is ready`))
```

Comment:
Yes, indeed, I could even add it as a 'debug log.' I will add a fix.

Reply:
I have added the fix in this PR:
https://github.com/nest-modules/mailer/pull/1121

And I have also added a function that returns true if all transporters are verified with no errors.

Reply:
@GFoniX use with https://www.npmjs.com/package/nodemailer-sendgrid says transporter.verify is not func

please test before PR

Reply:
@aecuto I apologize for this inconvenience. I have fixed this error in this pull request https://github.com/nest-modules/mailer/pull/1133. Indeed, I performed npm run test, but I did not manually test with SendGrid. Now, the verification is 'optional,' and I have still added an error check, just in case.